### PR TITLE
feat(canvas): perm syncing(4)

### PR DESCRIPTION
**Pull request by benwu408**
## Description

Implement retrieve_all_slim_docs_perm_sync for Canvas permission synchronization with batched SlimDocument yielding and heartbeat/stop-signal support. Add the EE get_course_permissions implementation that extracts enrolled user emails from the Canvas enrollments API with pagination. Register the Canvas connector in the frontend (icon, credential fields, source configuration). 

Includes unit tests for slim doc yielding, batching, permission caching, and all EE enrollment edge cases (email fallback, deduplication, pagination, API errors).

## How Has This Been Tested?

Unit tests covering retrieve_all_slim_docs_perm_sync (correct document IDs, external access attachment, batch sizing), permission caching (single call on repeated lookups, CE returns None), EE get_course_permissions (email extraction from login_id, fallback to email field, skip users without email, deduplication, pagination via next_url, empty enrollments, API error fallback), and canvas_doc_sync EE orchestrator (connector instantiated from connector_specific_config with decrypted credentials loaded, indexing_start forwarded as timestamp to the slim-doc fetch, None passed when no indexing_start is set, and empty dict used when credential_json is missing).          

Also tested on live canvas instance.
                                            
## PR Stack

**Stacked PRs on my fork (each targets the previous branch for isolated review):**
1. [#1 — Scaffold: API client, data models, connector registration](https://github.com/benwu408/onyx/pull/1)
2. [#2 — Connector data fetching and conversion](https://github.com/benwu408/onyx/pull/2)
3. [#3 — Checkpoint-based indexing](https://github.com/benwu408/onyx/pull/3)
4. [#4 — Permission sync and frontend registration](https://github.com/benwu408/onyx/pull/4)

**PRs on upstream onyx (all target main):**
1. [#9385 — Scaffold](https://github.com/onyx-dot-app/onyx/pull/9385)
2. [#9386 — Connector data](https://github.com/onyx-dot-app/onyx/pull/9386)
3. [#9387 — Checkpoint](https://github.com/onyx-dot-app/onyx/pull/9387)
4. [#9388 — Perm sync + frontend](https://github.com/onyx-dot-app/onyx/pull/9388)

---
## Summary by cubic
Adds Canvas permission sync using course enrollments and registers Canvas in the app. The sync yields batched `SlimDocument` IDs for pages, assignments, and announcements with per-course access, supports heartbeat/stop, and reports per-course progress.

- **New Features**
  - EE: `get_course_permissions` collects active enrollments (pagination, dedup), prefers `email` over `login_id`, ignores non-email `login_id` via an `@` check, lowercases, and returns private access when none found or on error; CE returns `None`.
  - Connector: `retrieve_all_slim_docs_perm_sync` emits per-course `SlimDocument` IDs with permissions; batching (batches can mix types); yields nothing when no courses; stop-signal raises; exceptions propagate to avoid mass revokes; progress per course.
  - Orchestrator: `canvas_doc_sync` added and registered in `_SOURCE_TO_SYNC_CONFIG` for `DocumentSource.CANVAS`; builds `CanvasConnector`, loads credentials, and forwards `indexing_start` as a timestamp; no group sync for Canvas.
  - Frontend: Added Canvas icon, `canvas_base_url` config, `canvas_access_token` credential; Auto Sync enabled; docs link at `/connectors/canvas`.

- **Bug Fixes**
  - Tests: fixed perm-sync mock to accept keyword args for `get_course_permissions` to match the call signature.

<sup>Written for commit 9ddb72641d90c0267f193e5c24031a5f05e192ac. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/onyx-dot-app/onyx/pull/10443?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: backend/ee/onyx/external_permissions/canvas/access.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    try:
+        for page in canvas_client.paginate(
+            f"courses/{course_id}/enrollments",
+            params={"per_page": "100", "state[]": "active", "include[]": "email"},
+        ):
+            for enrollment in page:
+                user = enrollment.get("user", {})
+                email = user.get("email") or user.get("login_id")
+                if email and "@" in email:
+                    emails.add(email)
```
Comment (jasmine-wu-ai) [inline backend/ee/onyx/external_permissions/canvas/access.py]: can you add some comments for these conditions? why can the email show up in two fields? it seems like we do some validation that this is real email but its kind of brittle?

Comment (benwu408) [reply backend/ee/onyx/external_permissions/canvas/access.py]: email is primary email field, but only works when viewer has permission to view email(which they should since it would be admin) and they have one set. also use login_id since most places use email as login too, the @ later is to make sure that if we did use the login_id it is an email.

**Review by justin-tahara — APPROVED** [review]
